### DISCUSSION TITLE
Is it possible to use locally subscribe method to catch events produced in different process?

### DISCUSSION THREAD

**Discussion by MoniaMoni** [RailsEventStore/rails_event_store#1410] [Q&A] [answered]
Hello,
I was reading RES documentation and Domain Driven Rails and I've learned from them much about event sourcing and rails_event_store, but I don't find answer on that question.
I would like implement SSE stream witch send data to client when specific event occur in system. A problem that I've got is that when I initialize subscriber locally in controller it doesn't see events produced on queue only thous one which appear in current process.
Is it possible to use RES like this kind subscriber? Or should I use other tool for this purpose?

**Comment by mpraglowski** [comment, marked as answer]
RES subscriptions works in scope of the process that RES client is running. You could handle that by implementing a catchup subscription pattern (reading from some stream/streams) in a separate process. See https://github.com/RailsEventStore/rails_event_store/tree/master/contrib/ruby_event_store-outbox outbox implementation from contrib projects. Maybe this will be suitable for you or be an inspiration how to implement your own solution.

**Reply by MoniaMoni** [reply]
Thank you for such a quick answer and advice.
